### ISSUE TITLE
error: Keras-Preprocessing 1.0.3 is installed but keras-preprocessing==1.0.2 is required by {'keras'}

### ISSUE THREAD

**Issue by bwanglzu** [tensorflow/tensorflow#22486]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**: pypi
- **TensorFlow version (use command below)**: tensorflow 1.11.0rc2
- **Python version**: 3.6
- **Bazel version (if compiling from source)**: not using
- **GCC/Compiler version (if compiling from source)**: not using
- **CUDA/cuDNN version**: not using
- **GPU model and memory**: not using
- **Exact command to reproduce**: Trying to install keras on Travis-CI (linux-Ubuntu 16.04), report: 

### Describe

Installed /home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/PyYAML-4.2b4-py3.6-linux-x86_64.egg
error: Keras-Preprocessing 1.0.3 is installed but keras-preprocessing==1.0.2 is required by {'keras'}

After I manually downgrade keras-preprocessing to 1.0.2 in requirements.txt and setup.py, it says:

error: Keras-Preprocessing 1.0.2 is installed but keras-preprocessing==1.0.3 is required by {'tensorflow'}

### Error Logs

Situation 1: keras-preprocessing 1.0.3 installed: [log](https://travis-ci.org/NTMC-Community/MatchZoo/jobs/429012836).

**Comment by moeppe-sp** [comment]
Same Problem for me and almost the same result with Keras Applications (tf needs 1.0.5 and Keras need 1.0.4)
I have no idea how to work my way around it

**Comment by beew** [comment]
https://github.com/tensorflow/tensorflow/issues/22601

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by bwanglzu** [comment]
anyone has solutions?

**Comment by facaiy** [comment]
@gunan The bug will be fixed in the coming 1.12 version, right?

**Comment by gunan** [comment]
The issue was keras_preprocessing was depending on keras. It is fixed in the latest version of keras_preprocessing.
While our dependency version is updated, you can always install TF with `--no_deps` and then install the latest version of keras_preprocessing.
Or you can simply install 1.12.0rc1 with `--pre` flag to `pip install tensorflow`

**Comment by bwanglzu** [comment]
thanks
